### PR TITLE
OHRM5X-1658: Develop dashboard - employee distribution by subunit widget

### PR DESCRIPTION
**Pull request by Super-Chama**

### CODE REVIEW

File: src/plugins/orangehrmDashboardPlugin/Controller/DashboardDataMockController.php [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        if (!$this->companyStructureService instanceof CompanyStructureService) {
+            $this->companyStructureService = new CompanyStructureService();
+        }
+        return $this->companyStructureService;
+    }
+
+    /**
+     * @return Response
+     */
+    public function getEmployeeSubunitDistribution(): Response
```
Comment (Super-Chama) [inline src/plugins/orangehrmDashboardPlugin/Controller/DashboardDataMockController.php:45]: @ManulMax Please check the json structure for your API task

File: src/client/src/orangehrmDashboardPlugin/components/EmployeeSubunitWidget.vue [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+      .then(response => {
+        const {data} = response.data;
+        const colors = Object.values(CHART_COLORS).reverse();
+        this.dataset = data.map((item, index) => {
+          return {
+            label: item.label,
+            value: item.employeeCount,
+            color:
+              //TODO: check lang string
+              item.label === 'Unassigned'
```
Comment (RajithaKumara) [inline src/client/src/orangehrmDashboardPlugin/components/EmployeeSubunitWidget.vue:81]: changed these data will come under meta. otherEmployeeCount & unassignedEmployeeCount.
Reason: if we need to localize `Unassigned`, can do in FE

File: src/plugins/orangehrmDashboardPlugin/Controller/DashboardController.php [hunk 3]
```diff
@@ -22,12 +22,29 @@
...
 {
+    use AuthUserTrait;
+
     public function preRender(Request $request): void
     {
-        $component = new Component('dashboard');
+        $component = new Component('view-dashboard');
         $this->setComponent($component);
+
+        // TODO: Rebase data group permisssions
```
Comment (RajithaKumara) [inline src/plugins/orangehrmDashboardPlugin/Controller/DashboardController.php:37]: TODO

File: src/plugins/orangehrmDashboardPlugin/Controller/DashboardDataMockController.php [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Dashboard\Controller;
+
+use OrangeHRM\Framework\Http\Response;
+use OrangeHRM\Core\Controller\AbstractController;
+use OrangeHRM\Admin\Service\CompanyStructureService;
+
+class DashboardDataMockController extends AbstractController
```
Comment (RajithaKumara) [inline src/plugins/orangehrmDashboardPlugin/Controller/DashboardDataMockController.php:27]: TODO::remove

File: src/plugins/orangehrmDashboardPlugin/config/routes.yaml [hunk 5]
```diff
@@ -5,9 +5,19 @@ apiv2_dashboard_employees_on_leave:
   defaults:
     _api: OrangeHRM\Dashboard\Api\EmployeeOnLeaveAPI
 
+##########################
+#       Mock API
+##########################
+
+apiv2_dashboard_employees_subunit_distribution:
```
Comment (RajithaKumara) [inline src/plugins/orangehrmDashboardPlugin/config/routes.yaml:12]: TODO::remove

**Review by RajithaKumara — APPROVED** [review]

File: src/client/src/orangehrmDashboardPlugin/components/EmployeeSubunitWidget.vue (hunk 2 above)
Comment (Super-Chama) [reply src/client/src/orangehrmDashboardPlugin/components/EmployeeSubunitWidget.vue:81]: done in https://github.com/orangehrm/orangehrm/pull/1428

File: src/plugins/orangehrmDashboardPlugin/config/routes.yaml (hunk 5 above)
Comment (Super-Chama) [reply src/plugins/orangehrmDashboardPlugin/config/routes.yaml:12]: removed in https://github.com/orangehrm/orangehrm/pull/1428

**Pull request merged**